pkg/util/test/util.go

```diff
@@ -217,3 +225,190 @@ func ActivateCRDWhenCreated(fakeClient *fakeextclient.Clientset, crdName string)
 		return false, nil
 	})
 }
+
+// UninstallStorageCluster uninstall and wipe storagecluster from k8s
+func UninstallStorageCluster(cluster *corev1alpha1.StorageCluster) error {
+	var err error
+	cluster, err = k8s.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if cluster.Spec.DeleteStrategy == nil ||
+		(cluster.Spec.DeleteStrategy.Type != corev1alpha1.UninstallAndWipeStorageClusterStrategyType &&
+			cluster.Spec.DeleteStrategy.Type != corev1alpha1.UninstallStorageClusterStrategyType) {
+		cluster.Spec.DeleteStrategy = &corev1alpha1.StorageClusterDeleteStrategy{
+			Type: corev1alpha1.UninstallAndWipeStorageClusterStrategyType,
+		}
+		if _, err = k8s.Instance().UpdateStorageCluster(cluster); err != nil {
+			return err
+		}
+	}
+
+	return k8s.Instance().DeleteStorageCluster(cluster.Name, cluster.Namespace)
+}
+
+// ValidateStorageCluster validates a StorageCluster spec
+func ValidateStorageCluster(
+	cluster *corev1alpha1.StorageCluster,
+	timeout, interval time.Duration,
+) error {
+	var err error
+	cluster, err = k8s.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
We should fail even if it is not found. Otherwise all the following checks will fail or return a nil pointer error.